### PR TITLE
feat: warn against squashing

### PR DESCRIPTION
**Pull request by joeyAghion**
As suggested in https://github.com/artsy/README/issues/516#issuecomment-1664293269.

### CODE REVIEW

**Review by brainbicycle — APPROVED** [review]
nice, thank you !

**Comment by joeyAghion** [comment]
This is 🍏 at long last. 😅

File: app/services/deploy_service.rb [hunk 1]
```diff
@@ -61,7 +61,7 @@ def create_pull_request
       deploy_strategy.arguments['base'],
       deploy_strategy.arguments['head'],
       'Deploy',
-      'This is an automatically generated release PR!'
+      "This is an automatically generated release PR!\n\nHumans: remember to _merge_ (not squash) this PR."
```
Comment (dblandin) [inline app/services/deploy_service.rb:64]: Should we update this to "(not squash or rebase)"? I believe we run into the same problem with rebase-merged deploy PRs.

Comment (joeyAghion) [reply app/services/deploy_service.rb:64]: I didn't realize that--sure!

**Comment by joeyAghion** [comment]
Gonna merge 🙇

**Pull request merged**